lib/std/mem.zig

```diff
@@ -196,26 +196,70 @@ test "Allocator.resize" {
 /// `copyForwards` if they do.
 pub const copy = copyForwards;
 
+/// Returns true if the slices overlap
+/// This is an *exclusive* range track
+pub fn slicesOverlap(comptime T: type, slice_a: []const T, slice_b: []const T) bool {
```

Comment:
Everything in std.mem already has to do with slices:

```suggestion
pub fn alias(comptime T: type, slice_a: []const T, slice_b: []const T) bool {
```